aws/resource_aws_rds_cluster_test.go

```diff
@@ -146,6 +146,60 @@ func TestAccAWSRDSCluster_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSRDSCluster_AllowMajorVersionUpgrade(t *testing.T) {
+	var dbCluster1, dbCluster2 rds.DBCluster
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_rds_cluster.test"
+	// If these hardcoded versions become a maintenance burden, use DescribeDBEngineVersions
+	// either by having a new data source created or implementing the testing similar
+	// to TestAccAWSDmsReplicationInstance_EngineVersion
+	engine := "aurora-postgresql"
+	engineVersion1 := "10.12"
+	engineVersion2 := "11.7"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSClusterConfig_AllowMajorVersionUpgrade(rName, false, engine, engineVersion1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSClusterExists(resourceName, &dbCluster1),
+					resource.TestCheckResourceAttr(resourceName, "allow_major_version_upgrade", "false"),
+					resource.TestCheckResourceAttr(resourceName, "engine", engine),
+					resource.TestCheckResourceAttr(resourceName, "engine_version", engineVersion1),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"allow_major_version_upgrade",
+					"apply_immediately",
+					"cluster_identifier_prefix",
+					"master_password",
+					"skip_final_snapshot",
+				},
+			},
+			{
+				Config:      testAccAWSClusterConfig_AllowMajorVersionUpgrade(rName, false, engine, engineVersion2),
+				ExpectError: regexp.MustCompile(`InvalidParameterCombination: The AllowMajorVersionUpgrade flag must be present when upgrading to a new major version`),
+			},
```

Comment:
looking at this more, would this test still be valid if we omit this step?  because at apply time, the `rds_instance` gets destroyed (as we'd expect) while the `rds_cluster` attempts to update, which ultimately errors out with this `ExpectError` value, but then this leaves us without a recreated `rds_instance`. so in the subsequent step, at plan time, we expect to create an `instance` and modify the `cluster`, but these 2 steps happen sequentially as the `instance` references the `cluster`, leaving the `cluster` to first be modified which then ultimately fails b/c of the missing primary instance :/

Reply:
unless maybe adding some `lifecycle` handling like `create_before_destroy` on the `instance` resource was a workaround to this step's config?

Reply:
Your first comment was right here. 👍  We actually do not want to have the `aws_rds_cluster_instance` to respond to `engine_version` changes at all (e.g. `ignore_changes`), preventing the recreation problem you mentioned on both errant and real engine version updates. Updating the documentation and test configuration should help.